app/lib/features/home/providers/navigation.dart

```diff
@@ -41,23 +42,51 @@ final spaceItemsProvider = FutureProvider.autoDispose
         ),
         location: '/$roomId',
       ),
-      data: (info) => SidebarNavigationItem(
-        icon: SpaceParentBadge(
-          roomId: roomId,
-          child: ActerAvatar(
-            uniqueId: roomId,
-            displayName: info.displayName,
+      data: (info) => canonicalParent.maybeWhen(
+        data: (parent) => SidebarNavigationItem(
+          icon: ActerAvatar(
             mode: DisplayMode.Space,
-            avatar: info.getAvatarImage(),
+            avatarInfo: AvatarInfo(
+              uniqueId: roomId,
+              displayName: info.displayName,
+              avatar: info.getAvatarImage(),
+            ),
+            avatarsInfo: parent != null
+                ? [
+                    AvatarInfo(
+                      uniqueId: parent.space.getRoomIdStr(),
+                      displayName: parent.profile.displayName,
+                      avatar: parent.profile.getAvatarImage(),
+                    ),
+                  ]
+                : [],
             size: 48,
           ),
+          label: Text(
+            info.displayName ?? roomId,
+            style: Theme.of(context).textTheme.labelSmall,
+            softWrap: false,
+          ),
+          location: '/$roomId',
         ),
-        label: Text(
-          info.displayName ?? roomId,
-          style: Theme.of(context).textTheme.labelSmall,
-          softWrap: false,
+        orElse: () => SidebarNavigationItem(
+          icon: ActerAvatar(
+            mode: DisplayMode.Space,
+            avatarInfo: AvatarInfo(
+              uniqueId: roomId,
+              displayName: info.displayName,
+              avatar: info.getAvatarImage(),
+            ),
+            avatarsInfo: const [AvatarInfo(uniqueId: '!')],
```

Comment:
what is this?

Reply:
An error fallback case, but its not needed now we will favor ```canonicalParent.valueOrNull```.